### ISSUE TITLE
can't find variable 'Fp_tar'

### ISSUE THREAD

**Issue by ruoqi77** [Liusifei/UVC#6]
In model.py, line 187:  self.grid_flat_crop = create_flat_grid(Fp_tar.size()).permute(0,2,1).detach()
I can't find the declaration of the variable 'Fp_tar', I wonder what the variable refers to?
Thanks a lot !

**Comment by sunshineatnoon** [comment]
Can you successfully run the test code?

**Comment by ruoqi77** [comment]
> Can you successfully run the test code?

Yes, the test code can be run successfully.  But all test codes use model 'Model_switchGTfixdot_swCC_Res'. The code I mentioned above is in 'track_match_comb', which is used in training phase. I haven't tried training yet, I just wonder what the variable refers to, cause it seems to appear out of nowhere. 
Thanks for your quick reply!

**Comment by sunshineatnoon** [comment]
Hi, I checked the code, the orthogonal constraint is from [line 178](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L178), the code from [line 184-191](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L184) is never reached (because `self.coord_switch` is always False, it's something we tried during experiments and should be removed). Sorry for the confusion, please let me know if this explanation makes sense.

**Comment by ruoqi77** [comment]
> Hi, I checked the code, the orthogonal constraint is from [line 178](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L178), the code from [line 184-191](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L184) is never reached (because `self.coord_switch` is always False, it's something we tried during experiments and should be removed). Sorry for the confusion, please let me know if this explanation makes sense.

OK, I get it. Thank you so much for your explanation !

**Comment by sunshineatnoon** [comment]
> > Hi, I checked the code, the orthogonal constraint is from [line 178](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L178), the code from [line 184-191](https://github.com/Liusifei/UVC/blob/5942e43e9c2e2107ecdd92865f02bfc1534fc9a7/model.py#L184) is never reached (because `self.coord_switch` is always False, it's something we tried during experiments and should be removed). Sorry for the confusion, please let me know if this explanation makes sense.
> 
> OK, I get it. Thank you so much for your explanation !

Awesome! close the issue now, please feel free to reopen if further discussion is needed.
